fenix/app/src/androidTest/java/org/mozilla/fenix/experimentintegration/GenericExperimentIntegrationTest.kt

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.experimentintegration
+
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mozilla.fenix.ext.settings
+import org.mozilla.fenix.helpers.HomeActivityTestRule
+import org.mozilla.fenix.helpers.TestHelper
+import org.mozilla.fenix.ui.robots.homeScreen
+
+class GenericExperimentIntegrationTest {
+    private val experimentName = "Viewpoint"
+
+    @get:Rule
+    val activityTestRule = HomeActivityTestRule(
+        isJumpBackInCFREnabled = false,
+        isPWAsPromptEnabled = false,
+        isTCPCFREnabled = false,
+        isDeleteSitePermissionsEnabled = true,
+    )
+
+    @Before
+    fun setUp() {
+        TestHelper.appContext.settings().showSecretDebugMenuThisSession = true
+    }
+
+    @After
+    fun tearDown() {
+        TestHelper.appContext.settings().showSecretDebugMenuThisSession = false
+    }
+
+    fun disableStudiesViaStudiesToggle() {
```

Comment:
Bit confused on this function. Is this a private function or a test? If the later it needs the `@Test` annotation and `test` prefix in the name, and a verification check.

Reply:
Private, will be called from the python code though, not from within the kotlin tests